Assets/Scripts/PlayerAnimator.cs

```diff
@@ -21,23 +25,44 @@ public void PlayFootSteps()
     // Update is called once per frame
     void Update()
     {
-        if (Input.GetKey(KeyCode.W) || Input.GetKey(KeyCode.S) || Input.GetKey(KeyCode.A) || Input.GetKey(KeyCode.D))
+        if (!isWalking)
         {
-            _playerAnimator.SetBool("isWalking", true);
-
-            if (Input.GetKey(KeyCode.LeftShift))
+            if (pressedWalking())
             {
-                _playerAnimator.SetFloat("walkingSpeed", 2f);
-            } else
-            {
-                _playerAnimator.SetFloat("walkingSpeed", 1f);
+                if (Random.Range(0f, 1.0f) < 0.5f)
+                {
+                    _starting_foot = "isWalking";
+                }
+                else
+                {
+                    _starting_foot = "isWalkingRight";
+                }
+                _playerAnimator.SetBool(_starting_foot, true);
+
+                if (Input.GetKey(KeyCode.LeftShift))
+                {
+                    _playerAnimator.SetFloat("walkingSpeed", 2f);
+                }
+                else
+                {
+                    _playerAnimator.SetFloat("walkingSpeed", 1f);
+                }
+                isWalking = true;
             }
-            
-        }
-        else
+        } else
         {
-            _playerAnimator.SetFloat("walkingSpeed", 0f);
-            _playerAnimator.SetBool("isWalking", false);
+            if (!pressedWalking())
+            {
+                _playerAnimator.SetFloat("walkingSpeed", 0f);
+                _playerAnimator.SetBool(_starting_foot, false);
+                isWalking = false;
+            }
         }
+        
+    }
+
+    bool pressedWalking()
+    {
+        return Input.GetKey(KeyCode.W) || Input.GetKey(KeyCode.S) || Input.GetKey(KeyCode.A) || Input.GetKey(KeyCode.D);
```

Comment:
```suggestion
    private bool pressedWalking()
    {
        return Input.GetKey(KeyCode.W) || Input.GetKey(KeyCode.S) || Input.GetKey(KeyCode.A) || Input.GetKey(KeyCode.D);
```

Reply:
Might be better to switch to just detecting movement with the Controller incase we change the controls or allow users to change key binds, maybe something like
```
Vector3 move = new Vector3(Input.GetAxis("Horizontal"), 0, Input.GetAxis("Vertical"));
if(move != Vector3.zero){
..
}
 ```
 Reference: https://docs.unity3d.com/ScriptReference/CharacterController.Move.html

Reply:
@wang-henry @emilyyyzh  we've switched away from the old Unity Input class to using the new Input System which allows us to enable and disable individual controls.

In this case
```
bool pressedWalking()
        return Input.GetKey(KeyCode.W) || Input.GetKey(KeyCode.S) || Input.GetKey(KeyCode.A) || Input.GetKey(KeyCode.D);
```

can be replaced with
```
public GameControls gameControls;
bool pressedWalking()
        return gameControls.Wrapper.Player.Move.IsPressed();
```
check our `CharacterMovement.cs` script for an example on what "GameControls" is, but in short, it's a scriptable object that wraps the new Unity Input System.